backend/routes/userRoutes.js

```diff
@@ -105,4 +105,142 @@ router.patch('/location', async(req, res) => {
   res.json(updatedUser)
 })
 
+router.get('/recommendedEvents/:userId', async (req, res) => {
```

Comment:
I would recommend to break this function into several, 1 for data retrieval, 1 for each score calculation, and 1 for data verification as Jerry mentioned

Reply:
Thank you.
I'll do a separate pr on that

Reply:
Done